make search not-regex by default

eeelnico opened the pull request:
for #219

interpret the `qry_string` parameter as just a string (not regex by default.

add an extra parameter `is_regex` to `orchestrator_api.query()`.
(this param is False for query(), but True in the context of `label_import_utils.get_element_group_by_texts()`)

DCO 1.1:
```
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

martinscooper commented:
Hi @eeelnico, thanks for taking this issue. 

Could you also add the boolean parameter in all the function calls all the way up to the [query API endpoint](https://github.com/label-sleuth/label-sleuth/blob/676bcabcbbe79fe0834cc6495eebd4b2998b5fbf/label_sleuth/app.py#L522). That would mean adding it to `orchestrator_api.query()`, `data_access.get_unlabeled_text_elements()` and `data_access.get_text_elements()`. This would allow the frontend to enable or disable using regex in the future. 

Some other suggestions: 
- changing `regex` to `isRegex` so the name implies that it is of type boolean and also adding the type, like `isRegex: bool` won't harm. 
- changing the query string variable name. Although in the function you worked with it is called `query`, in most of the other functions/methods it is called `query_regex`. I suggest to maintain `query` because now it can be a regex or not.

eeelnico commented:
> Hi @eeelnico, thanks for taking this issue.
> 
> Could you also add the boolean parameter in all the function calls all the way up to the [query API endpoint](https://github.com/label-sleuth/label-sleuth/blob/676bcabcbbe79fe0834cc6495eebd4b2998b5fbf/label_sleuth/app.py#L522). That would mean adding it to `orchestrator_api.query()`, `data_access.get_unlabeled_text_elements()` and `data_access.get_text_elements()`. This would allow the frontend to enable or disable using regex in the future.
> 
> Some other suggestions:
> 
>     * changing `regex` to `isRegex` so the name implies that it is of type boolean and also adding the type, like `isRegex: bool` won't harm.
> 
>     * changing the query string variable name. Although in the function you worked with it is called `query`, in most of the other functions/methods it is called `query_regex`. I suggest to maintain `query` because now it can be a regex or not.

thanks for your comments @martinscooper 
I pushed a new version , addressing your comments.
this is my last commit comment :
```
 address CR comments:

* rename 'query_regex' to just 'query'
* new param name is 'is_regex'
* add type (bool) and default (False) to new param in method declarations
* expose this param all the way up to OrchestratorAPI.query(): also DataAccessAPI and FileBasedDataAccess
```
(I used `is_regex` instead of `isRegex` just because python 🐍 )


comments:
* I replaced the old `query_regex` with the pair `query` + `is_regex` in 3 data-access-api methods:
  * `get_text_elements`
  * `get_labeled_text_elements`
  * `get_unlabeled_text_elements`
  but in OrchestratorApi
  * when it calls `get_text_elements()` it uses the query param
  * but when it calls `get_unlabeled_text_elements` it does not use the query
  * and the 3 times it calls `get_labeled_text_elements()` it also does not use them, but it's not even the context of query
* I saw tests failing when I pushed my first commit, how can I run the tests locally?

arielge commented on label_sleuth/app.py:
```diff
@@ -535,17 +535,16 @@ def query(workspace_id):
     query_string = request.args.get('qry_string')
```
Can you also modify the docstring of the `app.query()` endpoint?

eeelnico replied:
done, please verify, it said `regular expression`, I changed it to just `query string`

arielge reviewed (changes requested):
@eeelnico great job on handling the issue!

Once you have a [local environment](https://github.com/label-sleuth/label-sleuth#setting-up-a-development-environment), you can run the tests with `python -m unittest` from the repository root directory.
Please run the tests and make the necessary changes so that all tests will pass. Thanks!

arielge reviewed (approved):
Thanks @eeelnico, great work

arielge commented:
@eeelnico can you please add a copy of the [Developer's Certificate of Origin 1.1](https://elinux.org/Developer_Certificate_Of_Origin) to the PR description (as outlined in the [Contribution Guidelines](https://github.com/label-sleuth/label-sleuth/blob/main/CONTRIBUTING.md#contributor-license-agreement))? [Here](https://github.com/label-sleuth/label-sleuth.org/pull/12) you can see an example. Thanks!
